python/dlr/counter/model_metric.py

```diff
@@ -0,0 +1,70 @@
+import concurrent.futures
+import logging
+import time
+import json
+
+from .utils import resturlutils
+from . import config
+from .model_exec_counter import ModelExecCounter
+from .utils.helper import *
+
+class ModelMetric(object):
+    _pub_model_metric = True
+    _instance = None
+    MODEL_RUN = 3
+    resp_cnt = 0
+
+    @staticmethod
+    def get_instance(uuid):
+        """return single instance of class"""
+        if ModelMetric._instance is None:
+            ModelMetric._instance = ModelMetric(uuid)
+        return ModelMetric._instance
+
+    def __init__(self, uuid=0):
+        try:
+            self.client = resturlutils.RestUrlUtils()
+            self.uuid = uuid
+            # start loop
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=config.CALL_HOME_MAX_WORKERS_THREADS)
+            executor.submit(self.push_model_metric)
+            logging.info("model metric thread pool execution started")
+        except Exception as e:
+            logging.exception("model metric thread pool not started", exc_info=True)
+
+    def push_model_metric(self):
+        """publishing model run metric"""
+        while ModelMetric._pub_model_metric:
+            for key, val in ModelExecCounter.get_model_counts_dict().items():
+                self.model_run_info_publish(ModelMetric.MODEL_RUN, key, val)
+            ModelExecCounter.clear_model_counts()
+
+    def model_run_info_publish(self, model_event_type, model, count=0):
+        """push model load information at time model load time"""
+        try:
+            _md5model = get_hash_string(model.encode())
+            _md5model = str(_md5model.hexdigest())
+            pub_data = {'record_type': model_event_type, 'model': _md5model, 'uuid': self.uuid, 'run_count': count}
+            time.sleep(config.CALL_HOME_MODEL_RUN_COUNT_TIME_SECS)
```

Comment:
I wrote an integration test for this branch, and discover that this will cause the program to hang.  Please do make sure that this thread is exit properly when program ends.

Reply:
and what's the exact reason why we have a timeout here?